src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundUserAcceptanceTest.java

```diff
@@ -48,14 +49,32 @@ void tearDown() {
         System.setProperty("LAB_RESULTS_SCHEDULER_ENABLED", "false");
     }
 
-    @Test
-    void testSendEdifactIsProcessedAndPushedToGpOutboundQueue()
-        throws JMSException, IOException, InterchangeParsingException, MessageParsingException, JSONException {
-
-        final String content = new String(Files.readAllBytes(getEdifactResource().getFile().toPath()));
-
-        final OutboundMeshMessage outboundMeshMessage = OutboundMeshMessage.create(RECIPIENT,
-            WorkflowId.PATHOLOGY, content, null);
+    /**
+     * The test EDIFACT message (.dat file) is sent to the MESH mailbox where the adaptor receives inbound messages.
+     * The test waits for the messages to be processed and compares the FHIR message published to the GP Outbound Queue
+     * with the expected FHIR representation of the original message sent (.json file having the same name as the .dat)
+     */
+    @ParameterizedTest(name = "[{index}] - {0}")
+    @ArgumentsSource(SuccessArgumentsProvider.class)
+    void testEdifactIsSuccessfullyProcessedAndPushedToGpOutboundQueue(String category, TestData testData)
+            throws JMSException, InterchangeParsingException, MessageParsingException, JSONException {
+
+        final String recipient = getEdifactParser().parse(testData.getEdifact())
+            .getInterchangeHeader().getRecipient();
+
+        WorkflowId workflowId;
+        if (testData.getEdifact().contains(WorkflowId.PATHOLOGY.getWorkflowId())) {
```

Comment:
Are you checking if edifact string has workflow in it like anywhere? If yes, please use the edifact parser and check message header there

Reply:
Actually ignore that comment. It's ok for UAT purpose